lab3/client/src/views/generator.js

```diff
@@ -0,0 +1,74 @@
+import Generator from '../models/generator';
+
+
+export default function (store) {
+	return class GeneratorComponent extends window.HTMLElement {
+		constructor () {
+			super();
+			this.store = store;
+			this.onStageChange= this.handleStateChange.bind(this);
+			
+		
+		}
+	// TODO: subscribe to store on change event
+ 
+	handleStateChange (newState) {
+		const generator = new Generator(newState.generators[this.dataset.id]);
+		this.innerHTML = `<form class="card" action="">
+		<div class ="form group">
+				<div class = "rows">
+			<label for="Potion">${generator.name}</label>
+			<p class = 'holding'> ${generator.quantity}</p> 
+			</div>
+			<h5> ${generator.description}</h5>
+		</div>
+		<br>
+		<div class = "rows">
+				<p id ="B_Potion">${generator.rate}/10000</p>
+				<div class="actions">
+					<button type ="button">${generator.getCost()} Rupees</button>
+				</div>
+		</div>
+	</form>`
+		
+	}
+
+
+	connectedCallback (){	
+	const generator = new Generator(this.store.state.generators[this.dataset.id]);
+	this.innerHTML = `<form class="card" action="">
+		<div class ="form group">
+				<div class = "rows">
+			<label for="Potion">${generator.name}</label>
+			<p class = 'holding'> ${generator.quantity}</p> 
+			</div>
+			<h5> ${generator.description}</h5>
+		</div>
+		<br>
+		<div class = "rows">
+				<p id ="B_Potion">${generator.rate}/10000</p>
+				<div class="actions">
+					<button type = "button"> ${generator.getCost()} Rupees</button>
+				</div>
+		</div>
+	</form>`;
+	
+	this.addEventListener('click', () => {
```

Comment:
this binds to the whole generator instead of binding the click event to single button.

Reply:
not sure about this one , were we supposed to have it to something like 
document.querySelectorAll (#id of button )  instead ? i assumed this was fine since we were calling the generator 3 times just with different data sets